src/odb/src/db/dbTechLayer.cpp

```diff
@@ -1401,6 +1401,21 @@ int dbTechLayer::getSpacing()
   return layer->_spacing;
 }
 
+int dbTechLayer::getTrackSpacing(odb::dbTrackGrid* track_grid)
```

Comment:
This appears to assume a single track pattern.  In a layer like m2 in asap7 we have more than one track pattern and the spacing may be non-uniform.  I don't think this matters for grt because it usually uses a high layer but if we are making a general purpose API then this is somewhat misleading.

Reply:
Perhaps getAverageTrackSpacing might be more useful and add handling for the >1 track pattern case

Reply:
@gudeh Global router has an implementation for this. You can look at the `averageTrackPattern` function.